### PR TITLE
community: update OctoAI endpoint to subclass BaseOpenAI

### PR DESCRIPTION
**Pull request by sfvaroglu**
This PR updates OctoAIEndpoint LLM to subclass BaseOpenAI as OctoAI is an OpenAI-compatible service. The documentation and tests have also been updated.

### CODE REVIEW

File: libs/community/langchain_community/llms/octoai_endpoint.py [hunk 1]
```diff
@@ -1,166 +1,119 @@
...
+                client_params = {
+                    "api_key": values["octoai_api_token"].get_secret_value(),
+                    "base_url": values["octoai_api_base"],
+                    # To do: future support
+                    # "organization": values["openai_organization"],
+                    # "timeout": values["request_timeout"],
+                    # "max_retries": values["max_retries"],
+                    # "default_headers": values["default_headers"],
+                    # "default_query": values["default_query"],
+                    # "http_client": values["http_client"],
```
Comment (baskaryan) [inline libs/community/langchain_community/llms/octoai_endpoint.py:101]: can delete for now

File: libs/community/langchain_community/llms/octoai_endpoint.py [hunk 2]
```diff
@@ -1,166 +1,119 @@
...
+                top_p=0.9,
             )
 
     """
 
-    endpoint_url: Optional[str] = None
-    """Endpoint URL to use."""
-
-    model_kwargs: Optional[dict] = None
-    """Keyword arguments to pass to the model."""
```
Comment (baskaryan) [inline libs/community/langchain_community/llms/octoai_endpoint.py:51]: removing these is a breaking change, could we add some handling for them in the root validator?

File: libs/community/langchain_community/llms/octoai_endpoint.py (hunk 1 above)
Comment (sfvaroglu) [reply libs/community/langchain_community/llms/octoai_endpoint.py]: done

**Comment by sfvaroglu** [comment]
@baskaryan Do you mind merging this PR?

**Comment by sfvaroglu** [comment]
@baskaryan

**Review by baskaryan — APPROVED** [review]

**Pull request merged**